### PR TITLE
Feature/swagger20

### PR DESCRIPTION
**Pull request by adamhathcock**
Swagger 2.0 support https://github.com/khellang/Nancy.Swagger/issues/57

This isn't backwards compatible with 1.2.  I don't believe it really could live side by side. Likely a bit messy as well.  Getting more eyes on it would be good though.

### CODE REVIEW

File: samples/Nancy.Swagger.Demo/Nancy.Swagger.Demo.csproj [hunk 1]
```diff
@@ -126,6 +128,13 @@
   <PropertyGroup>
     <PreBuildEvent>xcopy "$(SolutionDir)lib\swagger-ui\dist" "$(ProjectDir)swagger-ui" /S /I /Y</PreBuildEvent>
   </PropertyGroup>
+  <Import Project="$(SolutionDir)\.nuget\NuGet.targets" Condition="Exists('$(SolutionDir)\.nuget\NuGet.targets')" />
+  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
+    <PropertyGroup>
+      <ErrorText>This project references NuGet package(s) that are missing on this computer. Enable NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+    </PropertyGroup>
+    <Error Condition="!Exists('$(SolutionDir)\.nuget\NuGet.targets')" Text="$([System.String]::Format('$(ErrorText)', '$(SolutionDir)\.nuget\NuGet.targets'))" />
```
Comment (khellang) [inline samples/Nancy.Swagger.Demo/Nancy.Swagger.Demo.csproj]: This is a problem. The old-style package restore has been enabled, and that's not something that we want :cry: To quote @davidebbo:

> Every Time someone enables nuget package restore on a solution, a kitten dies. Learn the new workflow!

http://blog.davidebbo.com/2014/01/the-right-way-to-restore-nuget-packages.html

**Comment by khellang** [comment]
Woah! Thanks for this! :+1: I'll have to take some time going through this though. First thing; we need to remove the old way of doing package restores. You could use [this script](https://github.com/owen2/AutomaticPackageRestoreMigrationScript). See http://www.xavierdecoster.com/migrate-away-from-msbuild-based-nuget-package-restore.

**Comment by adamhathcock** [comment]
Ew, sorry.  I didn't notice the old nuget thing.  Reverted.

Take your time.  I just wanted to get it out there.

**Comment by timschlechter** [comment]
:smiley: :+1:

**Comment by adamhathcock** [comment]
@timschlechter It would be possible.  I guess I was following the removal set out initially.  If backwards compatibility is highly desired then it's going to take more reworking.  Mainly figuring out namespacing then trying to marry the metadata building at some point.

Possibly, they don't need to be married at all.

The 2.0 has a large enough difference in the object model that I believe the code paths have to be nearly completely difference.  Things like Operations are now living under a common path when they didn't in 1.2.

I can give a stab at having them side-by-side but my first pass would likely not having them share much code.

**Comment by limanartem** [comment]
Hi guys,

Any plans for merging this PR and making it part of official nuget package?
I would be interested in using object model of swagger 2.0 in my project, so that looking forward to it!

Thanks
Artem

**Comment by adamhathcock** [comment]
My priorities changed but I still plan on using this.  I just haven't had the time to start putting back the old Swagger 1.2 code/objects along side the Swagger 2.0 stuff.

**Comment by felipeleusin** [comment]
Is it really necessary to get the Swagger 1.2 in this? We're picking this up at my company and want to get it out the door

**Comment by adamhathcock** [comment]
You could do it for it :)

It's not my call whether 1.2 still needs to exist or do something else to have side by side.

**Comment by felipeleusin** [comment]
Yeah, I'm already on it... so far this PR seems good. I'm taking a run using it to document an existing API we have here and see how it goes

File: src/Swagger.ObjectModel/Builders/SchemaBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+        #region Building
+
+        public SchemaBuilder<TModel> Discriminator(string discriminator)
+        {
+            this.discriminator = discriminator;
+            return this;
+        }
+
+        private SchemaBuilder<TModel> Description(string description)
```
Comment (masterpoi) [inline src/Swagger.ObjectModel/Builders/SchemaBuilder.cs]: Shouldn't Description be public here?

File: src/Nancy.Swagger/SwaggerConfig.cs [hunk 3]
```diff
@@ -8,8 +10,8 @@ public static class SwaggerConfig
         /// <summary>
         /// Contains all resolved ids by the <see cref="DefaultModelIdConvention"/>.
         /// </summary>
-        private static ConcurrentDictionary<string, Type> _resolvedModelIds = new ConcurrentDictionary<string, Type>();        
-        
+        private static ConcurrentDictionary<string, Type> _resolvedModelIds = new ConcurrentDictionary<string, Type>();
+
         /// <summary>
         /// The default resource listing path, <c>api-docs</c>.
```
Comment (masterpoi) [inline src/Nancy.Swagger/SwaggerConfig.cs]: In swagger 2.0 default path should be swagger.json, see https://github.com/swagger-api/swagger-spec/wiki/Swagger-1.2-to-2.0-Migration-Guide#whats-changed

**Pull request merged**